ci.py

```diff
@@ -0,0 +1,210 @@
+                if dep['name'] == 'global':
+                    dep['branch'] = branch_name
+        else:
+            global_branch = {'name': 'global', 'branch': branch_name}
+            self.custom_config = {'deps': [global_branch,]}
+
+        self.write_custom_config(self.custom_config)
+        repo.git.add(all=True)
+        repo.index.commit(f'Create feature build: {branch_name}')
+        origin = repo.remote(name='origin')
+        origin.push()
+        logging.info('Branch was created')
+        logging.info(f'Need to create PR now: https://github.com/Percona-Lab/pmm-submodules/compare/{branch_name}?expand=1')
+
+    def get_deps(self, single_branch=False):
+        with open(GIT_SOURCES_FILE, 'w+') as f:
+            f.truncate()
+
+        with open(GIT_SOURCES_FILE, 'a') as f:
+            for dep in self.config['deps']:
+                path = os.path.join(self.rootdir, dep["path"])
+                if not os.path.exists(os.path.join(self.rootdir, path)):
+                    target_branch = dep['branch']
+                    target_url = dep["url"]
+                    if single_branch:
+                        check_call(f'git clone --depth 1 --single-branch --branch {target_branch} {target_url} {path}'.split())
+                    else:
+                        check_call(f'git clone --depth 1 --no-single-branch {target_url} {path}')
+                else:
+                    print('Path for {} already exist'.format(dep["name"]))
+                call(["git", "pull", "--ff-only"], cwd=path)
+                commit_id = switch_or_create_branch(path, dep['branch'])
+
+                f.write(f'export {dep["name"]}_commit={commit_id}'.replace('-', '_'))
+                f.write(f'export {dep["name"]}_branch={dep["branch"]}\n'.replace('-', '_'))
+
+    def create_release(self):
+        pass
+    def create_tags(self):
+        pass
+
+class Converter():
+    def __init__(self, origin=SUBMODULES_CONFIG, target=YAML_CONFIG):
+        self.origin = origin
+        self.target = target
+        self.submodules = self.get_list_of_submodules()
+        self.convert_gitmodules_to_yaml()
+
+
+    def get_list_of_submodules(self):
+        config = configparser.ConfigParser()
+        config.read(self.origin)
+
+        submodules = []
+        for s in config.sections():
+            submodules_name = s.split('"')[1]
+            submodules_info = dict(config.items(s))
+            submodules_info['name'] = submodules_name
+
+            submodules.append(submodules_info)
+        return {'deps': submodules }
+
+    def convert_gitmodules_to_yaml(self):
+        yaml_config = Path(self.target)
+        if yaml_config.is_file():
+            logging.warning('File {} already exist!'.format(self.target))
+            sys.exit(1)
+        with open(self.target, 'w') as f:
+            yaml.dump(self.submodules, f, sort_keys=False)
+        sys.exit(0)
+
+def switch_or_create_branch(path, branch):
```

Comment:
we shouldn't create branch in case of some mistakes in branch name, so it's better to show error instead of silently creating new branch.